src/area.c

```diff
@@ -0,0 +1,115 @@
+#include "ui.h"
+#include "napi_utils.h"
+#include "control.h"
+#include "events.h"
+
+static const char *MODULE = "Area";
+
+struct area_handle {
+	/*
+		event registered on this control
+	*/
+	struct event_t *event_draw;
+	struct event_t *event_mouse;
+	struct event_t *event_mouseCrossed;
+	struct event_t *event_dragBroken;
+	struct event_t *event_key;
+};
+
+static void event_draw_cb(uiAreaHandler *h, uiArea *a, uiAreaDrawParams *p) {
+	// https://github.com/charto/nbind/issues/20
+	// struct event_t *event = (struct event_t *)data;
+	// fire_event(event);
+	printf("event_draw_cb\n");
+}
+static void event_mouse_cb(uiAreaHandler *h, uiArea *a, uiAreaMouseEvent *e) {
+	// struct event_t *event = (struct event_t *)data;
+	// fire_event(event);
```

Comment:
Actually, fire_event does not pass any arguments to the callback. We'll have to change it in order to allow an array of napi_value, to pass uiAreaDrawParams, uiAreaMouseEvent etc.

Reply:
Didn't do much more, because these need to somehow be transformed into objects accessible from JS.

Reply:
I can write them... maybe start now, and end them tomorrow...

Reply:
Good job, my compliments!

Reply:
> Good job, my compliments!

Thank you for laying the groundwork!